activation/activation.go

```diff
@@ -228,7 +218,13 @@ func (b *Builder) Register(sig *signing.EdSigner) {
 	b.postStates.Set(sig.NodeID(), types.PostStateIdle)
 
 	if b.stop != nil {
-		b.startID(b.parentCtx, sig)
+		ctx, stop := context.WithCancel(context.Background())
+		prevStop := b.stop
+		b.stop = func() {
+			prevStop()
+			stop()
+		}
+		b.startID(ctx, sig)
 	}
 }
```

Comment:
TBH, I'd just simplify and forbid `Registering()` when `Builder` is already running. We don't use/need to support this

Reply:
I didn't want to accidentally break things and originally the `Register` interface was intended to be used dynamically:

https://github.com/spacemeshos/go-spacemesh/issues/5090
https://github.com/spacemeshos/go-spacemesh/issues/5085

it is true that we don't use it at the moment, but I didn't want to break it either. It would be weird if it works everywhere else but here 😅 we can always simplify later if we want to.